packages/compiler/src/types/message.ts

```diff
@@ -56,11 +57,11 @@ export type ToolCall = {
 interface IMessage {
   role: MessageRole
   content: MessageContent[]
+  [key: string]: unknown
```

Comment:
We acknowledge people can put whatever they want as an attribute

```javascript
<user something={{ foo: 'bar' }} something_else="hi_nice_to_meet_you" />
```

We do the same with `<text>` blocks. This is not documented but you can do this:
```markdown
<system>
  Read this book and ask user questins
  <text cache_control={{ type: 'ephimeral' }}>...THE_ENTIRE_BOOK</text>

  Something else no cached
</system>
```

This will result in 3 system messages the one with the book cached in Anthropic in this case.